src/cpp/client/client_context.cc

```diff
@@ -75,6 +76,10 @@ void ClientContext::set_call(grpc_call* call,
   }
 }
 
+std::unique_ptr<const AuthContext> ClientContext::auth_context() const {
+  return CreateAuthContext(call_);
```

Comment:
It cannot be created from set_call because that is too early (the auth_context in core is not filled until hitting client_auth_filter).